standard-cluster-services/src/main/java/org/terracotta/monitoring/PlatformService.java

```diff
@@ -54,4 +54,51 @@
    * Provides the raw config file from server startup
    */
   InputStream getPlatformConfiguration();
+
+  /**
+   * Stops the server if it is in any of the passive states
+   * <p>
+   * Based on the {@code restartMode} argument, the server should either bounce or just halt (exit with correct exit codes).
+   * Currently, it is assumed that the halt may be abrupt and it may not clean up or empty any queues.
+   *
+   * @param restartMode mode in which the system needs to restart. STOP_ONLY means halt without restarting
+   */
+  void stopPlatformIfPassive(RestartMode restartMode);
+
+  /**
+   * Stops the server if it is in any of the active states (including active suspended)
+   * <p>
+   * Based on the {@code restartMode} argument, the server should either bounce or just halt.
+   * Currently, it is assumed that the halt may be abrupt and it may not clean up or empty any queues.
+   *
+   * @param restartMode mode in which the system needs to restart. STOP_ONLY means halt without restarting
+   */
+  void stopPlatformIfActive(RestartMode restartMode);
+
+  /**
+   * Stops the server.
+   * <p>
+   * Based on the {@code restartMode} argument, the server should either bounce or just halt (exit with correct exit codes).
+   * Currently, it is assumed that the halt may be abrupt and it may not clean up or empty any queues.
+   *
+   * @param restartMode mode in which the system needs to restart. STOP_ONLY means halt without restarting
+   */
+  void stopPlatform(RestartMode restartMode);
+
+  /**
+   * Returns true if it is safe to stop.
+   * <p>
+   * {@code partialStop} is true means this is a rolling stop (for rolling upgrades etc). Otherwise, this is a full stripe shutdown
+   * <p>
+   * If partialStop is true and if this server is ACTIVE_COORDINATOR, then return true iff all other stripe members are in PASSIVE_STANDBY, false otherwise
+   * If partialStop is true and if this server is PASSIVE_STANDBY, then return true
+   * If partialStop is true and if this server is any other state, then return false
+   * If partialStop is false, return true always (for now)
+   *
```

Comment:
Is this necessary?  It is not giving you good information since it is incredibly transient.  Anything in this area should be rolled into the operations above where the implementation can properly deal with it.

Reply:
partialStop true means it is only part of the stripe that is going down. In a planned shutdown case, it is important we fail the shutdown in case there is a possibility that system is currently not in a state (e.g mirror is not ready) to shutdown yet.

Reply:
Understood but this check is not helpful as the state can change right after check, thus the next call taking place on information that is no longer true.  Any check like this should occur in implementation under one of the shutdown calls added above.

Reply:
ok..Yes that makes absolute sense. (currently since we halt a step after the check, the same hole may still exist..but having an integrated API makes it easier to close that window in the future).

@ramsai1729 can you integrate the stop above with this..

  boolean stopPlatform(RestartMode restartMode, boolean partialMode);

if partialMode is true and condition is not safe to shutdown, return false...
otherwise just halt..

Reply:
@ramsai1729 Myself and @myronkscott  had a discussion..

So can you remove the isSafe... api  completely. 

There is no need of changes elsewhere including in stopPlatform...So ignore my comment above..

Throw an IllegalStateException if you cannot stop for the stop* calls. (so they can remain void)...

Reply:
done